src/Okta.Sdk/Api/UserApi.cs

```diff
@@ -2415,8 +2416,15 @@ public async System.Threading.Tasks.Task<Okta.Sdk.Client.ApiResponse<User>> GetU
             
             if (Sdk.Client.Configuration.IsPrivateKeyMode(this.Configuration) && !localVarRequestOptions.HeaderParameters.ContainsKey("Authorization"))
             {
-                var token = await _oAuthTokenProvider.GetAccessTokenAsync(cancellationToken: cancellationToken);
-                localVarRequestOptions.HeaderParameters.Add("Authorization", "Bearer " + token);
+                var tokenResponse = await _oAuthTokenProvider.GetAccessTokenResponseAsync(cancellationToken: cancellationToken);
+                localVarRequestOptions.HeaderParameters.Add("Authorization", $"{tokenResponse.TokenType} {tokenResponse.AccessToken}");
+
+                if (tokenResponse.IsDpopBound)
+                {
+                    var requestAbsoluteUri = new Uri(new Uri(this.Configuration.OktaDomain, UriKind.Absolute), new Uri($"/api/v1/users/{userId}", UriKind.Relative));
+                    var dPopProofJwt = _oAuthTokenProvider.GetDPopProofJwt(accessToken: tokenResponse.AccessToken, uri: requestAbsoluteUri.ToString(), httpMethod: "GET");
+                    localVarRequestOptions.HeaderParameters.Add("DPoP", dPopProofJwt);
+                }
```

Comment:
This logic should be delegated to the oAuthTokenProvider.

Reply:
@bryanapellanes-okta I followed the existing pattern provided by the codegen tooling, where headers and any request params are set by the corresponding APIs. If you take a look at other Api classes you will see this pattern.

Reply:
I'm suggesting encapsulation into a class responsible for these details.  Patterns established by codegen tools aren't always best practice.

Reply:
I couldn't agree more that the patterns established by code-gen tools aren't always the best practice, but there's a side that's not visible here, and it's templates. Templates determine in some way how you do things with this design. APIs are responsible for setting request details such as headers, payload, URIs, etc., So, keeping that in mind, the OAuthTokenProvider is responsible for providing tokens, and the Apis will update the request accordingly.

That being said, let's meet and continue this discussion so we speed up the review process. I'll ping you later.